std/haxe/macro/Compiler.hx

```diff
@@ -176,15 +176,30 @@ class Compiler {
 
 		@param rec If true, recursively adds all sub-packages.
 		@param ignore Array of module names to ignore for inclusion.
+		       You can use `module*` with a * at the end for Wildcard matching
 		@param classPaths An alternative array of paths (directory names) to use to search for modules to include.
 		       Note that if you pass this argument, only the specified paths will be used for inclusion.
 		@param strict If true and given package wasn't found in any of class paths, fail with an error.
 	**/
 	public static function include( pack : String, ?rec = true, ?ignore : Array<String>, ?classPaths : Array<String>, strict = false ) {
+		var ignore_wildcard:Array<String> = [];
+		if(ignore != null) {
+			var i = ignore.length;
+			while (i-- > 0) {
+				if(StringTools.endsWith(ignore[i], "*")) {
+					ignore_wildcard.push(ignore[i].substr(0, ignore[i].length-1));
+					ignore.splice(i, 1);
```

Comment:
It's probably better to just construct two arrays instead of modifying one of them. You shouldn't modify while iterating anyway.